Examples/CaseStudies/SwiftUICaseStudies/04-Navigation-Lists-NavigateAndLoad.swift

```diff
@@ -10,13 +10,15 @@ private let readMe = """
 
 @Reducer
 struct NavigateAndLoadList {
+  @ObservableState
   struct State: Equatable {
     var rows: IdentifiedArrayOf<Row> = [
       Row(count: 1, id: UUID()),
       Row(count: 42, id: UUID()),
       Row(count: 100, id: UUID()),
     ]
-    var selection: Identified<Row.ID, Counter.State?>?
+    var selectedRowID: UUID?
+    var selection: Counter.State?
```

Comment:
While this makes the `EmptyReducer` stuff less awkward, it unfortunately makes the domain modeled less precisely. It's not ideal, but let's return to using `Identified` for now.

Reply:
Thanks for the code review. As you suggested, I proceeded with the refactor using `Identified`. However, I encountered an issue. When trying to apply the comment(use `sending` method) to the `selection` in `NavigationLink`, refactoring with `Identified` resulted in an inability to use the `sending` method.

``` swift
selection: $store.selection?.id.sending(\.setNavigation) // Compile Error
```

As a workaround, I had to implement the selection using a `get` and `set` approach.
If you have any suggestions for a better solution, I'd appreciate your feedback. Thank you

Reply:
Additionally, the original code `state.selection?.value = Counter.State(count: state.rows[id: id]?.count ?? 0)` was causing an Overlapping accesses error, so I assigned the value to a separate variable to resolve this.

```swift
case .setNavigationSelectionDelayCompleted:
    guard let id = state.selection?.id else { return .none }
    // Overlapping accesses to 'state', but modification requires exclusive access; consider copying to a local variable
    state.selection?.value = Counter.State(count: state.rows[id: id]?.count ?? 0)
    return .none
```